flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactory.java

```diff
@@ -129,78 +142,183 @@ public DynamicTableSource createDynamicTableSource(Context context) {
 
 		final DataType physicalDataType = context.getCatalogTable().getSchema().toPhysicalRowDataType();
 
+		final int[] keyProjection = FactoryUtil.createKeyFormatProjection(tableOptions, physicalDataType);
+
+		final int[] valueProjection = FactoryUtil.createValueFormatProjection(tableOptions, physicalDataType);
+
+		final String keyPrefix = tableOptions.getOptional(FactoryUtil.KEY_FIELDS_PREFIX).orElse(null);
```

Comment:
Could we add a validation that the declared key fields should be all prefixed with the keyPrefix?